lib/corefoundation.rb

```diff
@@ -1,13 +1,26 @@
-require 'ffi'
-require 'iconv' if RUBY_VERSION < "1.9"
+require "ffi" unless defined?(FFI)
 
-require 'corefoundation/base'
-require 'corefoundation/null'
-require 'corefoundation/string'
-require 'corefoundation/array'
-require 'corefoundation/boolean'
-require 'corefoundation/data'
-require 'corefoundation/dictionary'
-require 'corefoundation/number'
-require 'corefoundation/date'
-require 'corefoundation/extensions'
+require_relative "corefoundation/base_wrapper"
+require_relative "corefoundation/base"
+require_relative "corefoundation/null"
+require_relative "corefoundation/string"
+require_relative "corefoundation/array"
+require_relative "corefoundation/boolean"
+require_relative "corefoundation/data"
+require_relative "corefoundation/dictionary"
+require_relative "corefoundation/number"
+require_relative "corefoundation/date"
+
+require_relative "corefoundation/error"
+require_relative "corefoundation/extensions"
+require_relative "corefoundation/preferences"
+
+# REVIEW: Include patches from module
```

Comment:
Are we good/green to monkey patch core classes?

Reply:
I still have mixed feelings. I would rather move this as a class method in CF classes instead of patching core classes, but the only reason I don't mind keeping them is they make conversions simpler due to inference. 

@tas50 @fcheung What are your thoughts on these patches?

Reply:
PS: The patches were moved to the module and then included because it didn't seem quite right to open the core classes. So this approach to monkey patching seemed to generate more confidence in general IMO

Reply:
It seemed reasonable given that we’re just adding 1 method that is unlikely to clash. 

If you felt strongly about it then these days maybe refinements would be a good fit?

Reply:
@rishichawda Irrespective of the approach to monkey patch, the concern is around the outcome of monkey patching core classes. I recommend checking this with @tas50  and other infra members.

Reply:
We could probably keep the patches for now and revisit this later.

Reply:
We have to get this reviewed by infra team before merging this